components/connector/internal/api/certificate_resolver.go

```diff
@@ -52,17 +52,19 @@ func NewCertificateResolver(
 }
 
 func (r *certificateResolver) SignCertificateSigningRequest(ctx context.Context, csr string) (*externalschema.CertificationResult, error) {
+	r.log.Debug("Authenticating the call for signing the Certificate Signing Request.")
+
 	clientId, err := r.authenticator.Authenticate(ctx)
 	if err != nil {
-		r.log.Errorf(err.Error())
+		r.log.Error("Failed authentication during the signing CSR process. ", err.Error())
```

Comment:
If its not possible to address with this PR, please create a separate issue to properly handle errors and remove the double error handling